RCS file: /cvs/gnome/gtk+/modules/input/gtkimcontextxim.c,v
retrieving revision 1.53
diff -u -p -r1.53 gtkimcontextxim.c
--- modules/input/gtkimcontextxim.c	30 Jun 2005 18:42:44 -0000	1.53
+++ modules/input/gtkimcontextxim.c	24 Mar 2006 07:26:53 -0000
@@ -756,6 +756,17 @@ gtk_im_context_xim_filter_keypress (GtkI
 
       buffer[num_bytes] = '\0';
 
+      /* If the buffer is good enough to be insert to other widgets,
+       * we will detect the modifiers state and decide whether to discard
+       * this buffer, before things become worse.
+       *
+       * Maybe this fix can't work on some platforms. Need be confirmed.
+       *
+       */
+      if (event->state &
+         (gtk_accelerator_get_default_mod_mask () & ~GDK_SHIFT_MASK))
+         return FALSE;
+
       result_utf8 = mb_to_utf8 (context_xim, buffer);
       if (result_utf8)
 	{
